### PR TITLE
fix night mode bug

### LINKED ISSUE #671: forced UI theme bug in 8.1 Oreo
i started using the 8.1 Pixel Experience rom instead of MIUI 9 on my Xiaomi Redmi 4x phone, and before, i could toggle between the light and the dark UI from the app's settings. now the system's option overwrites Tusky's theme - i can change to dark or light UI from the Tusky settings, but nothing will happen, only when i change the overall theme in the Android settings. Screenshots: https://imgur.com/a/kW58g45

* * * *
- Tusky Version: 1.8.0
- Android Version: 8.1
- Android Device: Xiaomi Redmi 4X
- Mastodon instance (if applicable): mastodon.social

- [ x ] I searched or browsed the repo’s other issues to ensure this is not a duplicate.

### PR DESCRIPTION
**Pull request by connyduck**
fixes #671

https://developer.android.com/reference/android/app/UiModeManager#setNightMode(int) 
>Developers interested in an app-local implementation of night mode should consider using `AppCompatDelegate.setDefaultNightMode(int)` to manage the -night qualifier locally.

Creating this PR so Bitrise creates a build that people experiencing this issue can confirm this fix. @bernd289 @nadszeg

On my devices this works as before.

### CODE REVIEW

**Comment by connyduck** [comment]
oh no it breaks tests

**Comment by connyduck** [comment]
> On my devices this works as before.

Nope getting a weird flicker when starting the app in light mode now. This needs more investigation.

File: app/src/main/java/com/keylesspalace/tusky/util/ThemeUtils.java [hunk 1]
```diff
@@ -121,12 +121,7 @@ public static void setAppNightMode(String flavor, Context context) {
                 break;
         }
 
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
```
Comment (bernd289) [inline app/src/main/java/com/keylesspalace/tusky/util/ThemeUtils.java]: Why did you remove the whole if instead of something like this:

```
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
            UiModeManager uiModeManager = (UiModeManager)context.getApplicationContext().getSystemService(Context.UI_MODE_SERVICE);
            uiModeManager.setNightMode(mode);
        }

        AppCompatDelegate.setDefaultNightMode(mode);
```

**Comment by mihudevcom** [comment]
Seems to be working fine for me. 

* Android Version: 9.0
* Android Device: Xiaomi Mi A1
* Mastodon instance (if applicable): mastodon.technology
